### PR TITLE
Move the ACN code to its new home in libs/acn

### LINKED ISSUE #966: Segfault from e131_transmit_test
Up to date as of d9722ce, configured with: 
```
./configure --prefix=/usr --enable-python-libs --enable-rdm-tests --disable-all-plugins --enable-usbpro --enable-e131
```

Steps to reproduce:
```
$ ./plugins/e131/e131/e131_transmit_test
```

Hit 'q' while it's running.

Example output: 
```
$ ./plugins/e131/e131/e131_transmit_test

========= E1.31 Tester ==========
------------------------------------
Test Case: 1/28
Test Name: Single Source Send
plugins/e131/e131/DMPE131Inflator.cpp:284: Raising priority for universe 1 from 0 to 20
plugins/e131/e131/DMPE131Inflator.cpp:299: Added new E1.31 source: e67753d2-e6d4-4735-88a8-3d5642692810
All tests passed.
Segmentation fault
```

Backtrace:
```
Program received signal SIGSEGV, Segmentation fault.
0x00000000006274d0 in ?? ()
(gdb) bt
#0  0x00000000006274d0 in ?? ()
#1  0x00007ffff78b05f6 in ola::io::StdinHandler::~StdinHandler (this=0x62efd0, __in_chrg=<optimized out>) at common/io/StdinHandler.cpp:60
#2  0x0000000000409787 in ~auto_ptr (this=0x7fffffffe978, __in_chrg=<optimized out>) at /usr/include/c++/4.9/backward/auto_ptr.h:170
#3  StateManager::~StateManager (this=0x7fffffffe900, __in_chrg=<optimized out>) at plugins/e131/e131/E131TestFramework.cpp:118
#4  0x00000000004071aa in main (argc=1, argv=0x7fffffffead8) at plugins/e131/e131/e131_transmit_test.cpp:275
```

Hope this helps.

### PR DESCRIPTION
**Pull request by nomis52**
Closes #966

### CODE REVIEW

**Comment by nomis52** [comment]
This is almost entirely transformed with sed commands.

File: libs/acn/Makefile.mk [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+endif
+
+# libolaacn.la
+libs_acn_libolaacn_la_SOURCES = \
+    libs/acn/CID.cpp \
+    libs/acn/CIDImpl.cpp \
+    libs/acn/CIDImpl.h
+libs_acn_libolaacn_la_CXXFLAGS = $(COMMON_E131_CXXFLAGS) $(uuid_CFLAGS)
+libs_acn_libolaacn_la_LIBADD = $(uuid_LIBS) \
+                                        common/libolacommon.la
```
Comment (peternewman) [inline libs/acn/Makefile.mk]: Fix the spacing

File: libs/acn/Makefile.mk [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    libs/acn/TCPTransport.h \
+    libs/acn/Transport.h \
+    libs/acn/TransportHeader.h \
+    libs/acn/UDPTransport.cpp \
+    libs/acn/UDPTransport.h
+
+libs_acn_libolae131core_la_CXXFLAGS = \
+    $(COMMON_E131_CXXFLAGS) $(uuid_CFLAGS)
+libs_acn_libolae131core_la_LIBADD = $(uuid_LIBS) \
+                                             common/libolacommon.la \
```
Comment (peternewman) [inline libs/acn/Makefile.mk]: Again

File: libs/acn/Makefile.mk [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                                             libs/acn/libolaacn.la
```
Comment (peternewman) [inline libs/acn/Makefile.mk]: Again

File: libs/acn/Makefile.mk [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+libs_acn_E133Tester_LDADD = \
+    libs/acn/libolae131core.la \
+    $(COMMON_TESTING_LIBS)
+
+libs_acn_TransportTester_SOURCES = \
+    libs/acn/TCPTransportTest.cpp \
+    libs/acn/UDPTransportTest.cpp
+libs_acn_TransportTester_CPPFLAGS = $(COMMON_TESTING_FLAGS)
+libs_acn_TransportTester_LDADD = libs/acn/libolae131core.la \
+                                          $(COMMON_TESTING_LIBS)
```
Comment (peternewman) [inline libs/acn/Makefile.mk]: Align

File: tools/e133/E133Endpoint.h [hunk 5]
```diff
@@ -57,7 +57,7 @@ class E133EndpointInterface
     // virtual bool rdm_enabled() const = 0;
 
     // ENDPOINT_MODE
-    // virtual ola::plugin::e131::EndpointMode endpoint_mode() const = 0;
```
Comment (peternewman) [inline tools/e133/E133Endpoint.h]: What's this for, extra stuff in the standard?

File: tools/e133/e133-receiver.cpp [hunk 6]
```diff
@@ -149,11 +149,11 @@ int main(int argc, char *argv[]) {
...
 
   // Setup E1.31 if required.
-  auto_ptr<ola::plugin::e131::E131Node> e131_node;
+  auto_ptr<ola::acn::E131Node> e131_node;
   if (FLAGS_e131) {
-    e131_node.reset(new ola::plugin::e131::E131Node(
+    e131_node.reset(new ola::acn::E131Node(
           node.SelectServer(), FLAGS_listen_ip,
-          ola::plugin::e131::E131Node::Options(), cid));
+          ola::acn::E131Node::Options(), cid));
```
Comment (peternewman) [inline tools/e133/e133-receiver.cpp]: Fix the spacing of this whole block.

**Comment by peternewman** [comment]
Update NEWS.

We should really fix the ordering of the includes too, but perhaps in a separate PR.

File: libs/acn/Makefile.mk (hunk 1 above)
Comment (nomis52) [inline libs/acn/Makefile.mk]: Done

File: libs/acn/Makefile.mk (hunk 2 above)
Comment (nomis52) [inline libs/acn/Makefile.mk]: Done

File: libs/acn/Makefile.mk (hunk 3 above)
Comment (nomis52) [inline libs/acn/Makefile.mk]: Done

File: libs/acn/Makefile.mk (hunk 4 above)
Comment (nomis52) [inline libs/acn/Makefile.mk]: Done

File: tools/e133/E133Endpoint.h (hunk 5 above)
Comment (nomis52) [inline tools/e133/E133Endpoint.h]: Yeah, stuff in the doc that I haven't implemented yet.

File: tools/e133/e133-receiver.cpp (hunk 6 above)
Comment (nomis52) [inline tools/e133/e133-receiver.cpp]: Done

File: tools/e133/E133Endpoint.h (hunk 5 above)
Comment (peternewman) [inline tools/e133/E133Endpoint.h]: Okay

**Comment by peternewman** [comment]
Oops, I think #996 may have been my fault.

Can you add something into NEWS, then LGTM. Although would be nice if we can look at the include ordering in a later PR.

**Comment by nomis52** [comment]
There is no NEWS item, this is just internal restructuring of the code.

**Pull request merged**